ruff_dev/Cargo.toml

```diff
@@ -16,3 +16,4 @@ rustpython-common = { git = "https://github.com/RustPython/RustPython.git", rev
 rustpython-parser = { features = ["lalrpop"], git = "https://github.com/RustPython/RustPython.git", rev = "c01f014b1269eedcf4bdb45d5fbc62ae2beecf31" }
 strum = { version = "0.24.1", features = ["strum_macros"] }
 strum_macros = { version = "0.24.3" }
+subprocess = "0.2.9"
```

Comment:
Just curious, why the dependency on this crate and not `std::process`?

Reply:
`std::process` doesn’t have the functionality of simultaneously writing to and reading from the child in a [deadlock-free way](https://docs.rs/subprocess/latest/subprocess/struct.Popen.html#method.communicate_start):

> The difference between this and simply writing input data to `self.stdin` and then reading output from `self.stdout` and `self.stderr` is that the reading and the writing are performed simultaneously. A naive implementation that writes and then reads has an issue when the subprocess responds to part of the input by providing output. The output must be read for the subprocess to accept further input, but the parent process is still blocked on writing the rest of the input daata. Since neither process can proceed, a deadlock occurs. This is why a correct implementation must write and read at the same time.

But actually, it looks like `rustfmt` always buffers all input before starting to write, so maybe `std::process` is fine.